rest_framework_dataclasses/serializers.py

```diff
@@ -307,6 +308,7 @@ def build_composite_field(self, definition: DataclassDefinition, field_name: str
         child_field = child_field_class(**child_field_kwargs)
         field_kwargs = {'allow_null': type_info.is_optional, 'child': child_field}
 
+        field_class: Type[SerializerField]
         if type_info.is_mapping:
             field_class = rest_framework.fields.DictField
         else:
```

Comment:
This is the kind of type declaration that seems unnecessary. Is mypy smart enough to infere the type of `field_class` if we make a ternary from it?

Reply:
It punts to just `type`, which is essentially `Type[Any]`. But it doesn't cause an error so I can keep it like this.
```python
        field_class = rest_framework.fields.DictField if type_info.is_mapping else rest_framework.fields.ListField
        reveal_type(field_class)  # Revealed type is 'builtins.type'
```